### DISCUSSION TITLE
Are BlackBox samples stored in microseconds as a 32 bit value ?

### DISCUSSION THREAD

**Discussion by mirov** [iNavFlight/inav#6785] [Q&A]
I've been trying to use the Blackbox logs to records flight params for very long fixed wing flights (1-2 hours).  So far the logs get truncated after about 70 min, with as it turns out is basically 2^32 microseconds.    Is this a limitation of the INAV side or the Blackbox explorer side ?

Is there a work around ?  I've set blackbox_rate_denom=200 so that the log files will fit into the dataflash of my FC.

**Comment by mirov** [comment]
Interestingly I remembered I had an old CleanFlight BB Explorer installed (Chrome app), and it does not show the truncation problem.  That's good news I suppose, it means that the flight logs are fine - it is just the visualization tools.

**Reply by stronnag** [reply]
Indeed, it's a limitation of your viewer.
```
$ flightlog2kml --summary /tmp/LOG00005.TXT
Log      : LOG00005.TXT / 1
Flight   : Miss Piggy on 2021-04-04 08:29:56
Firmware : INAV 2.7.0 (cb2fccb7) MATEKF405 of Feb 26 2021 19:16:33
Size     : 27.92 MB
Distance : 374 m
Duration : 81:27
Altitude : 6.3 m at 69:19
Speed    : 3.3 m/s at 03:58
Range    : 20 m at 00:22
Disarm   : Switch
```

81 minute bench test

**Reply by JulioCesarMatias** [reply]
this limitation should not exist, it is a mistake that it exists
